src/bootstrap/test.rs

```diff
@@ -444,7 +444,12 @@ impl Step for Miri {
                 builder.verbose(&format!("`cargo miri setup --env` returned: {:?}", stdout));
                 let sysroot = stdout.splitn(2, '=')
                     .nth(1).expect("`cargo miri setup` stdout did not contain '='");
-                sysroot.to_owned()
+                // This string is encoded to be eval'd by bash.
```

Comment:
we should totally find a better scheme than parsing arbitrary command line output.

Reply:
It's not "arbitrary", we control the other end of this. For parsing the actual path we could use https://docs.rs/snailquote/0.2.0/snailquote/fn.unescape.html. But I was not sure about adding another external dependency.

Or do you mean, we shouldn't be using the output mode designed for shell consumption here, because we are not a shell? I don't disagree. Instead of using `--env` here we could add a `--print-sysroot` flag that prints the plain sysroot without any escaping, `MIRI_SYSROT` or anything like that. Then we wouldn't have to parse here.

Reply:
> Or do you mean, we shouldn't be using the output mode designed for shell consumption here, because we are not a shell?

yes that